mandrill_test.go

```diff
@@ -82,6 +83,18 @@ func Test_MessagesSendTemplate_Fail(t *testing.T) {
 	expect(t, reflect.DeepEqual(correctResponse, err), true)
 }
 
+func Test_MessagesSendTemplate_Fail_Invalid_JSON(t *testing.T) {
+	server, m := testTools(400, ``)
+	defer server.Close()
+	responses, err := m.MessagesSendTemplate(&Message{}, "cheese", map[string]string{"name": "bob"})
+
+	expect(t, len(responses), 0)
+
+	correctResponse := json.Unmarshal([]byte(` `), struct{}{})
```

Comment:
`json.SyntaxError` has only `Offset` field exported, so this is the easiest way to test this case in current testing flow.